tiledb/sm/storage_manager/consolidator.cc

```diff
@@ -496,10 +488,27 @@ Status Consolidator::consolidate_fragment_meta(
   uint32_t meta_version = 0;
   RETURN_NOT_OK(utils::parse::get_fragment_version(meta_name, &meta_version));
 
+  // Calculate offset of first fragment footer
+  uint64_t offset = sizeof(uint32_t);  // Fragment num
+  for (auto m : meta) {
+    offset += sizeof(uint64_t);  // Name size
+    if (meta_version >= 9) {
+      offset += m->fragment_uri().last_path_part().size();  // Name
+    } else {
+      offset += m->fragment_uri().to_string().size();  // Name
+    }
+    offset += sizeof(uint64_t);  // Offset
+  }
+
   // Serialize all fragment names and footer offsets into a single buffer
   for (auto m : meta) {
     // Write name size and name
-    auto name = m->fragment_uri().to_string();
+    std::string name;
+    if (meta_version >= 9) {
```

Comment:
I do not think we need to condition on the version during the writes, or at least not based on the fragment metadata version. We may want to condition only based on the library version format for backwards write compatibility. In other words, for the newest library, we should always be writing relative paths. 

On the contrary, for the reads below, it is correct to condition on the fragment metadata version.

@joe-maley @Shelnutt2 thoughts?

Reply:
I don't think we can decide on this here. I have it on my personal TODO-list to write a compatibility requirements doc so that we can all discuss+agree on how to handle scenarios like this. 

I did a quick in this file for where we condition the write path based on the version. This only exists in the write path for null support (which I wrote). 

If we decide to always write as the latest version, this is a no-harm change. If we decide to support backward writes, this is useful. 

My vote is to leave this in for now. If we decide in the future that we don't need backwards compatible writes, I'll do a pass to remove this logic and my nullable logic in the write path.